xml/operating_system_linux.xml

```diff
@@ -0,0 +1,252 @@
+    <example os.version="5.11">SunOS 5.11</example>
+    <param pos="0" name="os.vendor" value="Sun"/>
+    <param pos="0" name="os.family" value="Solaris"/>
+    <param pos="0" name="os.product" value="Solaris"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:SUSE(?: SLED)?(?: Linux Enterprise Desktop)?\s?(\d+?(?:\.\d+?)*?)?)$">
+    <description>SUSE Linux Enterprise Desktop</description>
+    <example os.version="11">SUSE SLED 11</example>
+    <example os.version="12">SUSE Linux Enterprise Desktop 12</example>
+    <param pos="0" name="os.vendor" value="SUSE"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux Enterprise Desktop"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:SUSE(?: SLES)?(?: Linux Enterprise Server)?\s?(\d+?(?:\.\d+?)*?)?)$">
+    <description>SUSE Linux Enterprise Server</description>
+    <example os.version="11">SUSE SLES 11</example>
+    <example os.version="12">SUSE Linux Enterprise Server 12</example>
+    <param pos="0" name="os.vendor" value="SUSE"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux Enterprise Server"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:Ubuntu(?: Linux)?\s(\d+?(?:\.\d+?)*?)?\s?(LTS)?)$">
+    <description>Ubuntu Linux</description>
+    <example os.version="12.04.4">Ubuntu 12.04.4 LTS</example>
+    <example os.version="14.04">Ubuntu Linux 14.04</example>
+    <example os.version="16.04" os.edition="LTS">Ubuntu 16.04 LTS</example>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.edition"/>
+  </fingerprint>
+  <!-- Ubuntu derivative -->
+  <fingerprint pattern="^(?i:Xubuntu(?: Linux)?\s(\d+?(?:\.\d+?)*?)?\s?(LTS)?)$">
+    <description>Xubuntu Linux</description>
+    <example os.version="12.04.4">Xubuntu 12.04.4 LTS</example>
+    <example os.version="14.04">Xubuntu Linux 14.04</example>
+    <example os.version="16.04" os.edition="LTS">Xubuntu 16.04 LTS</example>
+    <param pos="0" name="os.vendor" value="Xubuntu"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.edition"/>
+  </fingerprint>
+  <!-- Linux catch-all goes at the bottom-->
+  <fingerprint pattern="^(?i:.*Linux?\s?(\d+?(?:\.\d+?)*?)?)$">
+    <description>Linux catch-all</description>
+    <example os.version="2.42.6">Linux 2.42.6</example>
+    <param pos="0" name="os.vendor" value="Linux"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <!-- other Unix-likes -->
```

Comment:
It's a argument for having a single file but yes I would either break it into more files or put it all in one.